### PR TITLE
[4.2] Add GitHub rebase script

### PR DESCRIPTION
**Pull request by HLeithner**
### Summary of Changes
Add a script to rebase pull requests

### Testing Instructions
Code review

### Documentation Changes Required
Add to internal docu

### CODE REVIEW

**Comment by HLeithner** [comment]
@brianteeman if you like to have a look at my english again. I used this script to rebase pull requests from 4.1-dev to 4.2-dev

File: build/github_rebase.php [hunk 1]
```diff
@@ -0,0 +1,153 @@
+<?php
+
+/**
+ * This script rebase Joomla Github Pull Requests to the target branch
```
Comment (brianteeman) [inline build/github_rebase.php]:
```suggestion
 * This script rebases Joomla Github Pull Requests to the target branch
```

File: build/github_rebase.php [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+ *
+ * @package            Joomla.Build
+ * @copyright      (C) 2022 Open Source Matters, Inc. <https://www.joomla.org>
```
Comment (brianteeman) [inline build/github_rebase.php]:
```suggestion
 * @package            Joomla.Build
 * @copyright          (C) 2022 Open Source Matters, Inc. <https://www.joomla.org>
```

File: build/github_rebase.php [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+		$cmd    = $gh . ' pr edit ' . $pr['url'] . ' --base ' . $targetBranch;
+		$output = [];
+		exec($cmd, $output, $result);
+		if ($result !== 0) {
+				var_dump([$cmd, $output, $result]);
+				die('Unable to set target branch for pr #' . $pr['number']);
+		}
+
+		$cmd    = $gh . ' pr comment ' . $pr['url'] . ' --body "This pull request has automatically rebased to ' . $targetBranch . '."';
```
Comment (brianteeman) [inline build/github_rebase.php]:
```suggestion
		$cmd    = $gh . ' pr comment ' . $pr['url'] . ' --body "This pull request has been automatically rebased to ' . $targetBranch . '."';
```

**Comment by brianteeman** [comment]
Sorry only just saw this. 

In addition to the inline comments what is clear from the results of running the script is that the comment in the pr needs some sort of additional message to show that the pr was created by a bot and not a human.

Also I assume that in the future this script and the psr12 script will be moved from your personal account to a dedicated bot account.

**Comment by HLeithner** [comment]
> Sorry only just saw this.
> 
> In addition to the inline comments what is clear from the results of running the script is that the comment in the pr needs some sort of additional message to show that the pr was created by a bot and not a human.
> 
> Also I assume that in the future this script and the psr12 script will be moved from your personal account to a dedicated bot account.

The account is used of the user who executes it, so psr12 will be run by me and the rebase by the release lead. Using the bot account for this doesn't makes it easier for us....

**Comment by HLeithner** [comment]
but hmm for the psr-12 PR updates I can try to use the bot... I will check try this

**Comment by brianteeman** [comment]
it should not be run from a personal account but from one that is dedicated to the purpose and _owned_ by the project. Its bad practice to use a private account for this.

**Comment by brianteeman** [comment]
LGTM

**Pull request merged**